### PR TITLE
validator: split out all library packages into their own .d.ts files

### PR DESCRIPTION
**Pull request by osdiab**
Because all the declarations are made in the root `validator/index.d.ts` file, if you don't automatically include the types from this library (i.e. don't define `types` in `tsconfig.json` `compilerOptions` field, or explicitly set `"types": ["validator"]`), then all the submodule types aren't accessible.

this puts all of them in their own files.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] **N/A** Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Similar issue from Material UI](https://github.com/mui-org/material-ui/issues/7853#issuecomment-324155653)
- [ ] **N/A** Increase the version number in the header if appropriate.
- [ ] **N/A** If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Lib files generated with this script:

```javascript
const path = require("path");
const fs = require("fs");

function findAll(regexPattern, sourceString) {
    let output = [];
    let match;
    // make sure the pattern has the global flag
    let regexPatternWithGlobal = RegExp(regexPattern, "g");
    while ((match = regexPatternWithGlobal.exec(sourceString))) {
        // get rid of the string copy
        delete match.input;
        // store the match data
        output.push(match);
    }
    return output;
}

const existing = fs.readFileSync("./index.d.ts").toString();
const regex = /declare module "validator\/(.+)" {[^}]+}/s;
const result = findAll(regex, existing);
result.forEach(r => {
    const [code, filename] = r;
    const parts = filename.split("/");
    const target = `${path.resolve(".", ...parts)}.d.ts`;
    const [declarationLine, constDeclaration, ...restOfCode] = code.split("\n");
    const importValidator = 'import * as validator from "validator";';
    const declareConstProperly = `declare ${constDeclaration.trim()}`;
    const toWrite = [
        importValidator,
        declareConstProperly,
        ...restOfCode.slice(0, -1).map(s => s.trim())
    ].join("\n");
    fs.writeFileSync(target, toWrite);
    console.log(`"${filename}.d.ts",`);
});
```

### CODE REVIEW

**Comment by osdiab** [comment]
BTW one thing I noticed is that not all the functions have definitions defined

**Review by kvcpr — APPROVED** [review]
LGTM

File: types/validator/lib/blacklist.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare module "validator/lib/blacklist" {
```
Comment (amcasey) [inline types/validator/lib/blacklist.d.ts]: Is the module declaration redundant now that the export is in a file with that path?

Comment (osdiab) [reply types/validator/lib/blacklist.d.ts]: Let me give it a shot!

Comment (osdiab) [reply types/validator/lib/blacklist.d.ts]: nope that wasn't happy lol

Comment (osdiab) [reply types/validator/lib/blacklist.d.ts]: says you can’t declare variables and exports outside of a declare block

**Comment by osdiab** [comment]
Updated so that each lib file doesn't declare its filename explicitly.

**Comment by amcasey** [comment]
Thanks!

**Pull request merged**